astropy/units/quantity.py

```diff
@@ -385,29 +385,31 @@ def __array_prepare__(self, obj, context=None):
                                         args[0].__class__.__name__,
                                         args[1].__class__.__name__))
 
-        # In the case of np.power, the unit itself needs to be modified by an
-        # amount that depends on one of the input values, so we need to treat
-        # this as a special case.
-        # TODO: find a better way to deal with this case
-        if function is np.power and result_unit != dimensionless_unscaled:
-
-            if units[1] is None:
-                p = args[1]
+        # In the case of np.power and np.float_power, the unit itself needs to
+        # be modified by an amount that depends on one of the input values,
+        # so we need to treat this as a special case.
+        # TODO: find a better way to deal with this.
+        if result_unit is False:
```

Comment:
I thought of using a string ('power'), but then one immediately runs into the problem that the unit starts to try to compare itself to that string. I guess in principle I could use another singleton (am using `None` already, so cannot do that...) or define one myself. But since I use `False` already in `converters`, I think I'll stick with it here....